scripts/spicy/zeek.spicy

```diff
@@ -147,54 +147,115 @@ public function protocol_handle_close(handle: ProtocolHandle): void &cxxname="ze
+public function file_gap(offset: uint64, len: uint64, fid: optional<string> = Null): void &cxxname="zeek::spicy::rt::file_gap";
 
 ## Signals the end of a file to Zeek's file analysis.
 ##
 ## fid: Zeek-side ID of the file to operate on; if not given, the file started by the most recent file_begin() will be used
-public function file_end(fid: optional<string> = Null) : void &cxxname="zeek::spicy::rt::file_end";
+public function file_end(fid: optional<string> = Null): void &cxxname="zeek::spicy::rt::file_end";
 
 ## Inside a packet analyzer, forwards what data remains after parsing the top-level unit
 ## on to another analyzer. The index specifies the target, per the current dispatcher table.
-public function forward_packet(identifier: uint32) : void &cxxname="zeek::spicy::rt::forward_packet";
+public function forward_packet(identifier: uint32): void &cxxname="zeek::spicy::rt::forward_packet";
 
 ## Gets the network time from Zeek.
-public function network_time() : time &cxxname="zeek::spicy::rt::network_time";
+public function network_time(): time &cxxname="zeek::spicy::rt::network_time";
+
+# TODO: The following types and functions still need documentation.
+
+public type ZeekRecord = __library_type("::zeek::spicy::rt::ValRecordPtr");
+public type ZeekSet = __library_type("::zeek::spicy::rt::ValSetPtr");
+public type ZeekTable = __library_type("::zeek::spicy::rt::ValTablePtr");
+public type ZeekVal = __library_type("::zeek::ValPtr");
+public type ZeekVector = __library_type("::zeek::spicy::rt::ValVectorPtr");
+
+public function get_address(id: string): addr &cxxname="zeek::spicy::rt::get_address";
+public function get_bool(id: string): bool &cxxname="zeek::spicy::rt::get_bool";
+public function get_count(id: string): uint64 &cxxname="zeek::spicy::rt::get_count";
+public function get_double(id: string): real &cxxname="zeek::spicy::rt::get_double";
+public function get_enum(id: string): string &cxxname="zeek::spicy::rt::get_enum";
+public function get_int(id: string): int64 &cxxname="zeek::spicy::rt::get_int";
+public function get_interval(id: string): interval &cxxname="zeek::spicy::rt::get_interval";
+public function get_port(id: string): port &cxxname="zeek::spicy::rt::get_port";
+public function get_record(id: string): ZeekRecord &cxxname="zeek::spicy::rt::get_record";
+public function get_set(id: string): ZeekSet &cxxname="zeek::spicy::rt::get_set";
+public function get_string(id: string): bytes &cxxname="zeek::spicy::rt::get_string";
+# public function get_subnet(id: string): network &cxxname="zeek::spicy::rt::get_subnet";
+public function get_table(id: string): ZeekTable &cxxname="zeek::spicy::rt::get_table";
+public function get_time(id: string): time &cxxname="zeek::spicy::rt::get_time";
+public function get_vector(id: string): ZeekVector &cxxname="zeek::spicy::rt::get_vector";
+
+public function get_value(id: string): ZeekVal &cxxname="zeek::spicy::rt::get_value";
```

Comment:
Maybe more general, we could return an `optional<ZeekVal>` here so users have a way to work around missing values instead of facing an exception.